fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/FixedDepositAccount.java

```diff
@@ -197,7 +197,7 @@ public void updateMaturityDateAndAmountBeforeAccountActivation(final MathContext
         String refNo = null;
         final Money transactionAmountMoney = Money.of(getCurrency(), this.accountTermAndPreClosure.depositAmount());
         final SavingsAccountTransaction transaction = SavingsAccountTransaction.deposit(null, office(), null,
-                this.accountSubmittedOrActivationDate(), transactionAmountMoney, refNo);
+                this.accountSubmittedOrActivationDate(), transactionAmountMoney, refNo, ExternalId.empty());
```

Comment:
We should check whether enable-auto-generated-external-id is true or false...

if it is true, we need to generate automatically external id.